### PR TITLE
Removes jQuery

### PR DESCRIPTION
**Pull request by gangadharjannu**
# PR is ready for functionality testing 

- [x] Removes jQuery in `jdd.js`
- [x] Tested

### CODE REVIEW

**Comment by zgrossbart** [comment]
I found a bug.  If you press the `Compare` button with nothing in either field you get this error in the browser console:

```
Uncaught TypeError: Cannot set properties of null (setting 'disabled')
    at Object.compare (jdd.js:1067:58)
    at HTMLButtonElement.<anonymous> (jdd.js:1147:13)
```

Then if you add some content and press the compare button again nothing happens.

**Comment by zgrossbart** [comment]
There are 21 jshint errors in `jdd.js`:

```
$ jshint jdd.js
jdd.js: line 679, col 103, Missing semicolon.
jdd.js: line 761, col 9, 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).
jdd.js: line 761, col 191, Missing semicolon.
jdd.js: line 867, col 69, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 947, col 33, Missing semicolon.
jdd.js: line 954, col 89, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 956, col 60, Missing semicolon.
jdd.js: line 959, col 93, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 961, col 64, Missing semicolon.
jdd.js: line 971, col 86, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 973, col 57, Missing semicolon.
jdd.js: line 976, col 90, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 978, col 61, Missing semicolon.
jdd.js: line 988, col 84, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 990, col 55, Missing semicolon.
jdd.js: line 993, col 88, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 995, col 59, Missing semicolon.
jdd.js: line 1024, col 74, Extra comma. (it breaks older versions of IE)
jdd.js: line 1028, col 23, Extra comma. (it breaks older versions of IE)
jdd.js: line 1030, col 35, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').
jdd.js: line 1145, col 45, Mixed double and single quotes.

21 errors
```

...and four in `jdd_test.js`

```
$ jshint jdd_test.js
jdd_test.js: line 270, col 27, Missing semicolon.
jdd_test.js: line 286, col 27, Missing semicolon.
jdd_test.js: line 304, col 50, ['string'] is better written in dot notation.
jdd_test.js: line 310, col 27, Missing semicolon.

4 errors
```

**Comment by zgrossbart** [comment]
Everything else looks good for the functionality.  I've tested this on the latest versions of Chrome, Firefox, Safari, and Edge.

File: jdd.js [hunk 1]
```diff
@@ -604,24 +604,19 @@ var jdd = {
...
-            codeBlockElement.addClass($(pre).attr('class'));
-            codeBlockElement.attr('id', $(pre).attr('id'));
-
-            $(pre).replaceWith(codeBlockElement);
+            var codeBlockElement = '<pre id="'+ pre.id +'" class="codeBlock ' + pre.classList.toString() + '">' + lineNumbers + codeLines + '</pre>';
+            pre.outerHTML = codeBlockElement;
         });
     },
 
+    // TODO: Not being used anywhere
```
Comment (zgrossbart) [inline jdd.js:619]: This is left over from an older implementation where I used to use a text area for the compare results.  I should probably delete it.

Comment (gangadharjannu) [reply jdd.js:619]: Shall I delete that in this PR or do that in a separate PR ?

**Comment by gangadharjannu** [comment]
Also in description I mentioned about What about `jsl.interactions.js` which is not used anywhere. Do we need that ?

If yes, I also need to convert that in native JavaScript

**Comment by gangadharjannu** [comment]
I added fixes for all the code review comments and pushed the changes

File: jdd.js (hunk 1 above)
Comment (zgrossbart) [reply jdd.js:619]: It's fine to delete this function as part of this PR, or leave it for another one.

**Comment by zgrossbart** [comment]
Let's leave the `jsl.interactions.js` file in the project for now.  I see that the reference has been removed from the `index.html` file so it isn't hurting anything.

**Review by zgrossbart — APPROVED** [review]
This is all looking good.  Thank you so much for the work.  I think this is a really positive change.  Please let me know when you're set and I'll merge it.

**Comment by gangadharjannu** [comment]
> This is all looking good. Thank you so much for the work. I think this is a really positive change. Please let me know when you're set and I'll merge it.

I also didn't remove jQuery in `jdd_test.js`. Do we need to do it ?

File: jdd.js (hunk 1 above)
Comment (gangadharjannu) [reply jdd.js:581]: I removed as part of this PR

**Comment by zgrossbart** [comment]
I'm fine leaving jQuery in the test page for now.  That file isn't used at runtime.  Let me know when you're done making changes and I'll merge it.

**Comment by gangadharjannu** [comment]
> I'm fine leaving jQuery in the test page for now. That file isn't used at runtime. Let me know when you're done making changes and I'll merge it.

I'm done changing :)

**Comment by zgrossbart** [comment]
You're all merged.  Thank you so much for all the work.  I'm going to do a little more testing and then I'll get it deployed.

**Pull request merged**